### ISSUE TITLE
Add support for CSS Variables in the Colour map

### ISSUE THREAD

**Issue by andrewnicols** [tinymce/tinymce#9788]
## 📝 Provide a description of the new feature or improvement
Add support for CSS Colours in the `color_map`, for example:

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  toolbar: 'forecolor backcolor',
  color_map: [
    'var(--black)', 'Black',
    'var(--gray)', 'Gray',
    'var(--white)', 'White',
    'var(--red)', 'Red',
    'var(--yellow)', 'Yellow',
    'var(--green)', 'Green',
    'var(--blue)', 'Blue'
  ]
});
```

### 🫶 What is the motivation?
The colour of content is intrinsically linked to the style of the page. If an organisation changes its branding, there should be a way to update content colouring accordingly. This also applies to content created on pages supporting a dark mode.

With support for CSS colour maps, it becomes possible for the page to override the colours for different themes, and cases such as dark mode (e.g. to invert colours).

### 🔗 What is the consequence of not having this feature?
Content colouring is hard-coded into the content, and difficult to manage and update.

### 🚦 How important would you rate the requested feature or improvement?
Critical.

We have a [tracker issue](https://tracker.moodle.org/browse/MDL-76517) for our product with a huge amount of interest. We have had to disable the colour picker until we can support updating content colouring in a maintainable fashion.

---

If you'd like to see this implemented sooner, add a 👍 reaction to this post.

**Comment by sluebbert** [comment]
At this point it may be worth a new issue to keep things clean, but I am very much interested in getting this feature to work for the background colors separately from the foreground colors.

The commits seem to just make the same `color_map_raw` assigned to both. I think what I'm looking for is a way to provide a `color_map_foreground_raw` and `color_map_background_raw`.

Let me know if this is outside the scope of this conversation and I can create a different one with the specific use case I have.

**Comment by TheSpyder** [comment]
Oh! It was released in TinyMCE 7.5 but we forgot to close this ticket. Yes, expanding it to separate background and foreground would be a new ticket.

There is more interest in this capability than I expected. Maybe we should've made some sort of object overload for color maps instead of adding a new `_raw` option... oh well, too late now. It wouldn't be difficult to make a new PR adding background and foreground raw options, similar to the PR @andrewnicols did, if you're interested.
